packages/flutter_tools/test/general.shard/build_system/targets/assets_transformers_test.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:args/args.dart';
+import 'package:file/memory.dart';
+import 'package:file_testing/file_testing.dart';
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:flutter_tools/src/base/user_messages.dart';
+import 'package:flutter_tools/src/build_system/build_system.dart';
+import 'package:flutter_tools/src/build_system/targets/assets.dart';
+import 'package:flutter_tools/src/cache.dart';
+
+import '../../../src/common.dart';
+import '../../../src/context.dart';
+import '../../../src/fake_process_manager.dart';
+
+void main() {
+  group('asset transformers', () {
+    final Artifacts artifacts = Artifacts.test();
+    final FileSystem fileSystem = MemoryFileSystem();
```

Comment:
I think these should all be initialized in the test. If someone adds a second test, they will easily leak state from the first test to the second.

Reply:
ahh, you want to set the filesystem and process manager as overrides...

Reply:
let's do the ugly thing, and initialize them in the overrides closure, and then access them via `globals.fs` from the test.

Reply:
Yeah, it didn't feel great doing this. IIRC one problem here is that there is a deep dependency on `FlutterProjectFactory` (when parsing the flutter manifest).